### PR TITLE
Fixed support activerecord 4.2.x + MySQL and Ruby 2.x

### PR DESCRIPTION
**Pull request by shiro16**
fixed
- `Hash#sort` ArgumentError: comparison of Array with Array failed
- `ActiveRecord::ConnectionAdapters::AbstractMysqlAdapter::Column#primary` not defined
- model require already defined error

### CODE REVIEW

File: lib/schema_doc/document.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ def model_classes
       end
 
       def model_classes_group_by_table_name
-        model_classes.group_by {|model_class|
+        model_classes.select {|model_class| model_class.table_exists? }.group_by {|model_class|
```
Comment (mizoR) [inline lib/schema_doc/document.rb]: I prefer `select(&:table_exists?)`, so could you add a commit?

File: lib/schema_doc/tasks.rake [hunk 2]
```diff
@@ -1,8 +1,12 @@
 namespace :schema_doc do
   task :out => :environment do
     require 'schema_doc'
-    Dir[Rails.root + 'app/models/**/*.rb'].each do |path|
-      require path
+    if defined?(Rails)
```
Comment (mizoR) [inline lib/schema_doc/tasks.rake]: If `Rails` was not defined then NameError will be occur L7.  So this condition check is not reasonable.

This gem is already depended on Rails now, so just `Rails.application.eager_load!` is OK for me 😃

How about you?

File: lib/schema_doc/document.rb (hunk 1 above)
Comment (shiro16) [reply lib/schema_doc/document.rb]: Oh! That's better.

File: lib/schema_doc/tasks.rake (hunk 2 above)
Comment (shiro16) [reply lib/schema_doc/tasks.rake]: Me, too!

File: lib/schema_doc/document.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@ class << self
       def read
         document = []
 
-        model_classes_group_by_table_name.sort.each do |table_name, model_classes|
+        model_classes_group_by_table_name.sort_by {|k, _| k }.each do |table_name, model_classes|
```
Comment (mizoR) [inline lib/schema_doc/document.rb]: I wonder if this change is required.  If `sort` was called without block, hash will maybe be sorted by key...?

``` irb
[1] pry(main)> RUBY_VERSION
=> "2.3.1"
[2] pry(main)> {"bob" => [100], "charlie" => [200], "alice" => [500]}.sort
=> [["alice", [500]], ["bob", [100]], ["charlie", [200]]]
```

**Comment by mizoR** [comment]
Released, thank you :) https://rubygems.org/gems/schema_doc/versions/0.0.2

**Comment by shiro16** [comment]
Thanks !

**Pull request merged**